core/azure-core/src/main/java/com/azure/core/util/polling/PollResponse.java

```diff
@@ -0,0 +1,150 @@
+     *
+     * <p><strong>Code Sample Creating PollResponse Object</strong></p>
+     * {@codesnippet com.azure.core.util.polling.pollresponse.status.value.retryAfter.properties}
+     *
+     * @param status Mandatory operation status as defined in {@link OperationStatus}
+     * @param value The value as a result of poll operation. This can be any custom user defined object.
+     * @param retryAfter How long poller should wait before calling next pollOperation.This can be null but negative or zero value is not allowed.
+     * @param properties The map of perperties which user might need in executing poll operation such as service url etc.
+     * This could be used by Poll Operation function when it gets called by {@link Poller}, since this {@link PollResponse} object
+     * will be supplied to poll operation in next polling cycle.
+     *
+     * @throws NullPointerException If {@code status} is {@code null}.
+     * @throws IllegalArgumentException if {@code  retryAfter} is negative or zero
+     */
+    public PollResponse(OperationStatus status, T value, Duration retryAfter, Map<Object, Object> properties) {
+        Objects.requireNonNull(status, "The status input parameter cannot be null.");
+        if (retryAfter != null && retryAfter.toNanos() <= 0) {
+            throw new IllegalArgumentException("Negative or zero value for retryAfter not allowed.");
+        }
+        this.status = status;
+        this.value = value;
+        this.retryAfter = retryAfter;
+        this.properties = properties;
+    }
+
+    /**
+     * Create a Poll Response with status, value and retryAfter
+     *
+     * <p><strong>Code Sample Creating PollResponse Object</strong></p>
+     * {@codesnippet com.azure.core.util.polling.pollresponse.status.value.retryAfter}
+     *
+     * @param status Mandatory operation status as defined in {@link OperationStatus}
+     * @param value The value as a result of poll operation. This can be any custom user defined object.
+     * @param retryAfter How long poller should wait before calling next pollOperation. This can be null but negative or zero value is not allowed.
+     */
+    public PollResponse(OperationStatus status, T value, Duration retryAfter) {
+        this(status, value, retryAfter, null);
+
+    }
+
+    /**
+     * Create a Poll Response with status and value
+     *
+     *<p><strong>Code Sample Creating PollResponse Object</strong></p>
+     * {@codesnippet com.azure.core.util.polling.pollresponse.status.value}
+     *
+     * @param status Mandatory operation status as defined in {@link OperationStatus}
+     * @param value The value as a result of poll operation. This can be any custom user defined object.
+     */
+    public PollResponse(OperationStatus status, T value) {
+        this(status, value, null);
+    }
+
+    /**
+     * Status of Long Running Operation at the time of polling. Also see {@link OperationStatus}
```

Comment:
@JonathanGiles   "completed successfully" : This might confuse with LRO completed as defined in OperationStatus Enum.  Would following be fine ?

"Represents the status of the long-running operation at the time the last polling operation finished successfully."

Reply:
That's fine.